### PR TITLE
Center AppBar title on iOS

### LINKED ISSUE #4962: iOS: center text in app header
On iOS, a common pattern is to center the text in the app header.

iphone_6s_plus_-_iphone_6s_plus___ios_9_3__13e230_

### PR DESCRIPTION
**Pull request by abarth**
This patch adapts the AppBar to feel more like iOS by centering the title.

Fixes #4962

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie @HansMuller

File: packages/flutter/lib/src/material/app_bar.dart [hunk 1]
```diff
@@ -210,17 +216,32 @@ class AppBar extends StatelessWidget {
...
+    if (centerTitle != null)
+      return centerTitle;
+    assert(themeData.platform != null);
+    switch (themeData.platform) {
+      case TargetPlatform.android:
+        return false;
+      case TargetPlatform.iOS:
+        return true;
+    }
+    return false;
```
Comment (Hixie) [inline packages/flutter/lib/src/material/app_bar.dart]: return null here (and the other one)

**Comment by Hixie** [comment]
test?

**Comment by Hixie** [comment]
LGTM

**Comment by abarth** [comment]
> test?

I wrote a test but looks like I didn't upload it.  :(

**Pull request merged**